src/items/generics.md

```diff
@@ -43,18 +43,137 @@ struct Ref<'a, T> where T: 'a { r: &'a T }
 struct InnerArray<T, const N: usize>([T; N]);
 ```
 
+The generic parameters are in scope within the item definition where they are
+declared.
+
+[References], [raw pointers], [arrays], [slices][arrays], [tuples], and
+[function pointers] have lifetime or type parameters as well, but are not
+referred to with path syntax.
+
+### Const generics
+
+Const generic parameters allow items to be generic over constant values. The
+const identifier introduces a name for the constant parameter, and all
+instances of the item must be instantiated with a value of the given type.
+
+<!-- TODO: update above to say "introduces a name in the [value namespace]"
+    once namespaces are added. -->
+
 The only allowed types of const parameters are `u8`, `u16`, `u32`, `u64`, `u128`, `usize`
 `i8`, `i16`, `i32`, `i64`, `i128`, `isize`, `char` and `bool`.
 
-Const parameters may only be be used as standalone arguments inside
-of [types] and [repeat expressions] but may be freely used elsewhere:
+Const parameters can generally be used anywhere a [const item] can be used,
+with the exception of the definition of any [item] within the body of a
```

Comment:
This exception applies to all generics. Can it be placed outside of the `const generics` section? Or we can point to [item declaration statements](https://doc.rust-lang.org/stable/reference/statements.html#item-declarations) as the source of truth. Moving it out of this section also lets us trim down this sentence:

> Const parameters can be used anywhere a const item can be used, with the exception that when used in a type or array repeat expression, it must be standalone (as described below). That is, they are allowed in the following places:

Reply:
I added a sentence at the top when talking about parameter scopes that links to _Item declaration statements_.